### PR TITLE
khal: add options for default and view sections

### PR DESCRIPTION
**Pull request by arjan-s**
### Description

The current khal module doesn't provide options to add settings in the [default] and [view] sections of the configuration file. This PR adds optional settings for both. I chose a generic option instead of separately defining all possible settings, due to the amount of them and the future maintenance required for keeping the list up to date.

### Checklist

- [x] Change is backwards compatible.

- [x] Code formatted with `./format`.

- [x] Code tested through `nix-shell --pure tests -A run.all` or `nix develop --ignore-environment .#all` using Flakes.

- [x] Test cases updated/added. See [example](https://github.com/nix-community/home-manager/commit/f3fbb50b68df20da47f9b0def5607857fcc0d021#diff-b61a6d542f9036550ba9c401c80f00ef).

- [x] Commit messages are formatted like

    ```
    {component}: {description}

    {long description}
    ```

    See [CONTRIBUTING](https://github.com/nix-community/home-manager/blob/master/docs/contributing.adoc#sec-commit-style) for more information and [recent commit messages](https://github.com/nix-community/home-manager/commits/master) for examples.

- If this PR adds a new module

  - [ ] Added myself as module maintainer. See [example](https://github.com/nix-community/home-manager/blob/068ff76a10e95820f886ac46957edcff4e44621d/modules/programs/lesspipe.nix#L6).

### CODE REVIEW

**Comment by arjan-s** [comment]
I don't know how to fix the macos test, sorry

File: modules/programs/khal.nix [hunk 1]
```diff
@@ -147,13 +147,41 @@ let
...
+
     locale = mkOption {
       type = lib.types.submodule { options = localeOptions; };
       description = ''
         khal locale settings. 
       '';
       default = { };
     };
+
+    defaultSection = mkOption {
```
Comment (rycee) [inline modules/programs/khal.nix:159]: Maybe just add a `settings`-option? Then people can do

```
settings = {
  default = {
    default_calendar = "calendar";
    timedelta = "5d";
  };
  view = {
    agenda_event_format = "{calendar-color}{cancelled}{start-end-time-style} {title}{repeat-symbol}{reset}";
  };
};
```

**Comment by rycee** [comment]
Oh and about the macos test, it is unrelated to this PR and I think it should resolve itself on your next push 🙂

File: modules/programs/khal.nix (hunk 1 above)
Comment (arjan-s) [reply modules/programs/khal.nix]: Sounds like a good plan, I didn't really like the two different-but-the-same sections anyway. And this approach also lets people configure other sections like [highlight_days] and [keybindings] too, so it's a double win.

**Comment by arjan-s** [comment]
Thanks! What do you think about it now?

**Comment by teto** [comment]
tested here and worked well

**Pull request merged**